### DISCUSSION TITLE
最近收到的录播数据错误

### DISCUSSION THREAD

**Discussion by HMBSbige** [HMBSbige/BilibiliLiveRecordDownLoader#70] [General]
2021-03-20 和 30号两天录播上传失败，显示`转码失败`，明明已经转成 mp4 且本地播放器能播放。以前这种情况出现的时候，往往是时间戳问题（转成 mp4 封装能解决绝大部分问题）。我也没多想，这两天的录播我就直接重新压制上传了...

结果从4月份开始连续 5、6 天的录播都出现同样情况。通过观察分P，都是1P出错，直接猜测开头接受到的数据问题。

果然，通过分析这几天开头接受到的数据（已排除网络环境，甚至用其他录播软件），都有一个共性：收到 FLV 头和 I帧前都有一段 4s 多的音频数据

甚至直播回放下载下来的 flv 也有同样的情况，不知道什么原因导致的，可能跟主播推流有关（主播用的是 OBS 推流，应该也不可能有问题）

本地播放不影响，只有上传 B 站会出现`转码失败`

目前的解决方法：
1. 重新压制（浪费时间）
2. 用本工具或 ffmpeg 剪掉前面 5s。会剪掉前面正常内容的 5s 和多余的 4s 多音频。剪掉的时间小于多余的音频时间会发现输出的文件大小相同。
3. 用本工具或 ffmpeg 剪掉后面的 1s。会剪掉后面正常内容的 1s 和多余的 4s 多音频。剪掉的时间甚至可以是 0.1s ，但过小会发现输出的文件大小相同。
4. 如果有其他工具可以很方便删掉前面多余的音频理论也可以

**Comment by SekiBetu** [comment]
剪开头就完事了
